### ISSUE TITLE
[Bug]: models parameter leads to weird end year vector dynamics in SSplotComparisons

### ISSUE THREAD

**Issue by kellijohnson-NOAA** [r4ss/r4ss#861]
### Describe the bug

The for loop [here](https://github.com/r4ss/r4ss/blob/adc6b67d949e39345edb7a21eff04e228adc4f2c/R/SSplotComparisons.R#L707C21-L707C29) in `SSplotComparisons()` leads to the incorrect end year being plotted for the models of interest if you are dropping any models. I believe the correct code is
```
for (iline in models) {
...
}
```
but I am not sure why you need to execute the code inside the loop for just the models that you want plotted. It seems like you could just do the assigning of NA values to all values beyond end year for ALL THE MODELS. It might save some coding and speed because it would be vectorized rather than looping through. Not sure on the speed part though.

### To Reproduce

```
output <- r4ss::SS_output(system.file(package="r4ss", "extdata", "simple_small"), verbose = FALSE, printstats = FALSE)
mod.sum <- r4ss::SSsummarize(list(output, output, output), verbose = FALSE)
r4ss::SSplotComparisons(
  mod.sum,
  models = c(1, 3),
  legendlabels = c("First model", "Third model"),
  subplot = 1,
  print = FALSE,
  plot = TRUE,
  endyr = c(2018, 2012, 2015)
)
```
The third model is truncated at 2012 when it should go to 2015 according to endyr.

### Expected behavior

I wanted the correct end year to be plotted when using SSplotComparisons(models = c(1,3)), i.e., not plotting model 2.

### Screenshots

![image](https://github.com/r4ss/r4ss/assets/4108564/cd3920d9-1ecc-4f63-ab3b-cb90d84e3f80)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/r4ss/r4ss#code-of-conduct)

**Comment by iantaylor-NOAA** [comment]
The intent was to have the length of `endyrvec` equal to the number of models plotted, not the total number of models before any subsetting. Thus, the default endyrvec is set for each of the `nlines` rather than for `models`:
https://github.com/r4ss/r4ss/blob/adc6b67d949e39345edb7a21eff04e228adc4f2c/R/SSplotComparisons.R#L694-L696.
This is also the case for other inputs like `col` and `lwd`.

In hindsight this setup is not as convenient for subsetting models because you have to modify endyrvec each time. But switching approaches could break any existing code that combines a vector of `endyrvec` values with a `models` input.

Would it be adequate to add an error when the `models` input is shorter than the `endyrvec` input and improve the documentation of `models` input and those other inputs that need to match the length of `models`?

**Comment by kellijohnson-NOAA** [comment]
Okay, that was not clear from the documentation. Maybe we can just change the default behavior of endyrvec if models has an input, which is what was leading to my troubles. The default is NULL 
https://github.com/r4ss/r4ss/blob/adc6b67d949e39345edb7a21eff04e228adc4f2c/R/SSplotComparisons.R#L197
and I would have thought that the default behavior wouldn't have side effects.

**Comment by iantaylor-NOAA** [comment]
@kellijohnson-NOAA, thanks for the additional info. I failed to realize that this issue occurred when the source models have different endyr values, not just when user inputs `endyrvec`.

I fixed it in 0ef80c1 pushed to main after checking the before/after in the plot below using the code.

I won't close this until after I improve the documentation.
```
output <- r4ss::SS_output(system.file(package="r4ss", "extdata", "simple_small"), 
  verbose = FALSE, printstats = FALSE)
mod.sum <- r4ss::SSsummarize(list(output, output, output), verbose = FALSE)
mod.sum$endyrs = c(2018, 2012, 2015)    
r4ss::SSplotComparisons(
      mod.sum,
      models = c(1, 3),
      lwd = c(3,5), 
      lty = c(1,3),
      spacepoints = 100,
      legendlabels = c("First model (endyr 2018)", "Third model (endyr 2015)"),
      subplot = 1,
      print = FALSE,
      plot = TRUE,
      new = FALSE
    )
```
![issue_861](https://github.com/r4ss/r4ss/assets/4992918/472b3aeb-850b-4c45-9877-71af1fbb5b21)

**Comment by iantaylor-NOAA** [comment]
Fixed via 0ef80c1 and b5fb1c1.
